opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -307,12 +307,16 @@ namespace Opm
                     
                     const bool hasGroupControl = this->isInjector() ? inj_controls.hasControl(Well::InjectorCMode::GRUP) :
                                                                       prod_controls.hasControl(Well::ProducerCMode::GRUP);
-                    bool isGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP; 
-                    if (! (isGroupControl && !this->param_.check_group_constraints_inner_well_iterations_)) {
-                        changed = this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
-                    }
-                    if (hasGroupControl && this->param_.check_group_constraints_inner_well_iterations_) {
-                        changed = changed || this->checkGroupConstraints(well_state, group_state, schedule, summary_state,deferred_logger);
+                    const bool isGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP;
+
+                    if (this->param_.check_group_constraints_inner_well_iterations_) {
+                        if (!isGroupControl && hasGroupControl) {
+                            changed = changed || this->checkGroupConstraints(well_state, group_state, schedule, summary_state,deferred_logger);
+                        }
+                        changed = changed || this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
```

Comment:
This one seems problematic though, since individual constraints will not be checked if 'changed' is true, i.e., if we violate a group constraint. Not sure if it is intentional? In practice it appears to cause issues with wells not respecting individual constraints while on group control...

Reply:
I have updated it accordingly. I am not sure what the best solution is. Only allow for one switch per iteration or always respect individual constraints.